design/ceph/osd-migration.md

```diff
@@ -38,137 +27,101 @@ considered in the future:
-    ```
-
-#### OSD Activation
-
-The ceph-volume `activate` command doesn't require the OSD backend to be passed
-as an argument. It auto-detects the backend that was used during when the OSD
-was prepared.
-
-### Existing OSDs
-
-- After upgrading an existing Rook Ceph cluster to a Ceph release that supports
-  `bluestore-rdr`, admins can migrate `bluestore` OSDs to `bluestore-rdr`.
-- The backend of OSDs can not be overridden, so this update will require
-  the OSDs to be replaced one by one.
-- In order to migrate OSDs to use `bluestore-rdr`, admins must patch the
-  Ceph cluster spec as below:
-
-```yaml
-storage:
-  store:
-    type: bluestore-rdr
-    updateStore: yes-really-update-store
-```
-
-#### Replace OSD
-The operator's reconciler will replace one OSD at a time.
-A configmap will be used to store the OSD ID currently being migrated.
 OSD replacement steps:
-  1. List all OSDs where `osd-store:<osd store type>` does not match `spec.storage.store.type`.
-  2. If all PGs are not `active+clean`, do not proceed.
-  3. If all PGs  are `active+clean` but a previous OSD replacement is not completed, do not proceed.
-  4. If all PGs are `active+clean` and no replacement is in progress, then select an OSD to be migrated.
-  5. Delete the OSD deployment.
-  6. Create an OSD prepare job with an environment variable indictating the OSD ID to be replaced.
-  7. The OSD prepare pod will destroy the OSD and prepare it again using the same OSD ID. Refer [Destroy OSD](#destroy-osd) for details.
-  8. Once the destroyed OSD pod is recreated, delete the configmap.
-  9. If there is any error during the OSD migration, then preserve the OSD ID being replaced in the configmap for next reconcile.
-  10. Reconcile the operator and perform as same steps until all the OSDs have migrated to the new backend.
-
-#### Destroy OSD
-
-The OSD prepare pod job will destroy an OSD using following steps:
-- Check for OSD ID environment variable of the OSD to be destroyed.
-- Use `ceph volume list` to fetch the OSD path.
-- Destroy the OSD using following command:
-
-    ```
-    ceph osd destroy <OSD_ID> --yes-i-really-mean-it
-    ```
-
-- Wipe the OSD drive. This removes all the data on the device.
-- Prepare the OSD with the new store type by using the same OSD ID. This is done by passing the OSD ID as `--osd-id` flag to `ceph-volume` command.
-
-    ```
-    ceph-volume lvm prepare --osd-id <OSD_ID> --data <OSD_PATH>
-    ```
-
-## Planning
-
-These changes require significant development effort to migrate existing OSDs to use a new backend.
-They will be divided into following phases:
-* New OSDs (greenfield).
-* Migrating existing OSDs on PVC without metadata devices.
-* Migrating existing OSDs on PVC with metadata devices.
-* Existing node-based OSDs (multiple OSDs are created at once via `ceph-volume batch` which adds additional complications).
+
+1. List all OSDs where `osd-store:<osd store type>` does not match `spec.storage.store.type`.
```

Comment:
are we planning to rework this to do this on a per-deviceset basis still?

Reply:
This example is of the OSD Store type. This is a global setting (in `spec.storage`) is applied for all the devicesets. But in case of encryption, which is per deviceset setting, migration will be done per-deviceset only.